src/Logging.AzureAppServices/src/BlobLoggerProvider.cs

```diff
@@ -55,15 +56,25 @@ public BlobLoggerProvider(IOptionsMonitor<AzureBlobLoggerOptions> options)
             _fileName = value.ApplicationInstanceId + "_" + value.BlobName;
             _blobReferenceFactory = blobReferenceFactory;
             _httpClient = new HttpClient();
+            _customPrefixFileName = value.CustomPrefixFileName;
         }
 
         internal override async Task WriteMessagesAsync(IEnumerable<LogMessage> messages, CancellationToken cancellationToken)
         {
             var eventGroups = messages.GroupBy(GetBlobKey);
+            DateTime currDate = DateTime.Now;
             foreach (var eventGroup in eventGroups)
             {
                 var key = eventGroup.Key;
-                var blobName = $"{_appName}/{key.Year}/{key.Month:00}/{key.Day:00}/{key.Hour:00}/{_fileName}";
```

Comment:
I tracked down the original change that used the old key format in https://github.com/aspnet/Logging/pull/637. However, it doesn't explain why that format is used. My main concern is that the new API actually changes the folder structure but my concerns are lessened somewhat because it's only when the new API is used and non of the previous behaviours are changed.

This change of format depending on whether a prefix is passed feels very strange and should at least clearly documented in the xml doc comments.

Reply:
Yes I hear you Jun Tao. Should we establish an enum that drives the output file structure? For example FileStructure.Flat for the new overload, and FileStructure.Hierarchy for the original functionality. Then the custom file prefix could be used for FileStructure.Flat only?